### PR TITLE
add ChromeScreenshotReportSource.{timeout,jsRunOnLoad}

### PR DESCRIPTION
**Pull request by speezepearson**
The ability to run JS on page-load is necessary to hook our reporting system into Grafana.

Having a timeout isn't necessary, merely prudent-seeming.

### CODE REVIEW

File: app/models/ebean/ChromeScreenshotReportSource.java [hunk 1]
```diff
@@ -43,6 +44,12 @@
     @Column(name = "ignore_certificate_errors")
     private boolean ignoreCertificateErrors;
 
+    @Column(name = "timeout")
+    private Duration timeout;
```
Comment (vjkoskela) [inline app/models/ebean/ChromeScreenshotReportSource.java]: I believe there is a problem with `Duration` serialization through EBean where it truncates the seconds. 

https://github.com/ArpNetworking/metrics-portal/pull/179/files#r279255056

File: app/models/internal/impl/ChromeScreenshotReportSource.java [hunk 2]
```diff
@@ -204,6 +252,10 @@ public Builder setIgnoreCertificateErrors(final Boolean ignoreCertificateErrors)
         @NotEmpty
         private String _triggeringEventName;
         @NotNull
+        private Duration _timeout;
```
Comment (vjkoskela) [inline app/models/internal/impl/ChromeScreenshotReportSource.java]: Probably needs to be positive as well.

File: test/java/com/arpnetworking/metrics/portal/TestBeanFactory.java [hunk 3]
```diff
@@ -146,6 +146,8 @@ private TestBeanFactory() {
                                 .setId(UUID.randomUUID())
                                 .setTriggeringEventName(TEST_EVENT + UUID.randomUUID().toString())
                                 .setUri(URI.create("http://" + UUID.randomUUID().toString().replace("-", "") + ".example.com"))
+                                .setTimeout(Duration.ofSeconds(123))
+                                .setJsRunOnLoad("some_javascript()")
```
Comment (vjkoskela) [inline test/java/com/arpnetworking/metrics/portal/TestBeanFactory.java]: Neither of these is random. For the timeout, it should probably be a reasonable value in a range (e.g. 1 sec to 10 sec). For the JS, it should probably pick from some set of valid JS constants including not setting any (using the default of empty string).

**Comment by speezepearson** [comment]
> This code does not seem to actually use either of these values. The timeout I can understand. What's the use case for the JS?

It's useful to be able to manipulate the page before taking a screenshot of it, in order to remove the visual clutter of the parts that don't matter -- for our Grafana use case, this means replacing the whole page-body with the rendered iframe's body.

**Comment by vjkoskela** [comment]
As discussed; use `&kiosk` on the URL to avoid any post-processing.

**Comment by speezepearson** [comment]
Taking a completely different approach. Abandoning this PR.

**Pull request closed without merging**